rfcs/dogfood-pull-request-sla-monitor.md

```diff
@@ -0,0 +1,126 @@
+# Dogfooding Wing - Pull Request SLA 
+
+- **Author(s):**: @ekeren
+- **Submission Date**: {2023-01-04}
+- **Stage**: Draft
+
+**Goal**: Have a real application in production that we develop with wing, debug it, extend its capabilities and monitor it
+
+## Reasoning For Dogfooding
+
+One of the p1 of our beta launch is having a real applications that we use in production,
+AKA as dogfooding, it is believed that without it: 
+> **We will not know what we don't know about the readiness of Wing**
+
+## This RFC scope
+
+This request for comments is focusing on a specific application **pull request SLA monitor**,
+that will be extended in the future for a community companion tool that can monitor 
+and respond to issues/pull requests/slack posts, etc... 
+
+The specific code presented here is just the initial work, it is not completed but I still believe I can 
+get valuable feedback from the team before I move forward. 
+
+There is also an emphasis here on not future proofing anything, or creating future abstractions.
+For example the slack is basicaly an alerting mechanism, but right now it is just plain old slack 
+
+```ts (wing)
+bring cloud;
+/** 
+* Using octokit as rest client for github api
+* I've decided not to create a github Resource because
+* I don't see a reason to mock this resource in the sim.
+* This is different from the slack resource, where I don't want to
+* go back and forth to slack and spam some channel in my developement mode.
+*/ 
+bring untyped octokit as octokit; // syntax TBD 
+
+/**
+* This is an initial, quick and dirty API of 
+* wing's equivilant to javascript setTimeout api
+* There are many options how the API for capability can look like
+* for example, It might be a capability of the cloud.Function object.
+* But I decided to focus on a designated Resource for this capability, 
+* so we can implement this as a custom resource and not an sdk resource (to begin with) 
+*/ 
+resource FutureCloudFunctionInvoker { 
+  /**
+  * Invoke an inflight function in the future
+  * @praram delay - when to lauch this 
+  * @param fn - which function to invoke
+  */
+  inflight invoke(delay: duration, fn: inflight () => void): void {
```

Comment:
Wondering if this should be called `on_schedule()`? (in the spirit of our other event-based APIs).

Reply:
Interesting... 

`scheduler.on_schedule(2h, do_something)`
 
Creating and canceling can probably be the minimum for this api, but it will probably eventually include creating, removing, listening-to-events-of and querying scheduled jobs. 

very similar to Java's ([Scheduler](https://docs.oracle.com/cd/E24152_01/Platform.10-1/apidoc/atg/service/scheduler/Scheduler.html), [ScheduledJob](https://docs.oracle.com/cd/E24152_01/Platform.10-1/apidoc/atg/service/scheduler/ScheduledJob.html))